### PR TITLE
Use reloadData for large updates

### PR DESCRIPTION
**Pull request by smeis**
Split of from #132 as per request @benlodotcom.

### CODE REVIEW

File: ComponentKit/DataSources/CKCollectionViewDataSource.mm [hunk 1]
```diff
@@ -290,14 +288,20 @@ static void applyChangesetToCollectionView(const Output::Changeset &changeset, U
...
-    [collectionView deleteItemsAtIndexPaths:itemRemovalIndexPaths];
-  }
-  if (itemUpdateIndexPaths.count > 0) {
-    [collectionView reloadItemsAtIndexPaths:itemUpdateIndexPaths];
-  }
-  if (itemInsertionIndexPaths.count > 0) {
-    [collectionView insertItemsAtIndexPaths:itemInsertionIndexPaths];
+  if(itemUpdateIndexPaths.count > 30){
+    [collectionView reloadData];
+  }else{
```
Comment (adamjernst) [inline ComponentKit/DataSources/CKCollectionViewDataSource.mm]: nit: spacing

**Comment by smeis** [comment]
Thanks @adamjernst, fixed!

**Comment by benlodotcom** [comment]
Can you give more details regarding why reloadData was needed ?

**Comment by adamjernst** [comment]
I think this is a fine optimization, but we want to keep the "sample" data source simple. I think the best thing for you to do is to fork the data source and put this optimization there!

**Comment by smeis** [comment]
I'm sorry for not getting back to this sooner @adamjernst @benlodotcom , I'm unusually busy right now. I had been working on building a sample project to clarify the problem but hadn't gotten around to finishing it and I don't really have time to finish it right now. Especially since it might take a lot more digging since I started suspecting this might only show up in pretty specific cases, so your suggestion is probably the best for now.

The problem was that when I reloaded the dataset to resize the components the app stalled the main thread. It spent a huge amount of time in _endAnimations as you can see in the screenshot below. I put in the call to reloadData because I remembered UICollectionView could sometimes crash if you enqueued too many updates at once. I expect to have more time to look at this in about a month or so.




ck-stall-screen-modified

**Pull request closed without merging**